rcl_lifecycle/CMakeLists.txt

```diff
@@ -11,6 +11,21 @@ find_package(rmw REQUIRED)
 
 include_directories(include)
 
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(NOT WIN32)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wpedantic")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
```

Comment:
I went ahead and revert to check the compiler instead of the operating system in https://github.com/ros2/rcl/pull/141/commits/2ecb36d556db75c1c99d07182e9ad5cc21e20939